pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go

```diff
@@ -85,141 +81,79 @@ func (tr *TrustedTaskRun) Validate(ctx context.Context) (errs *apis.FieldError)
 func (tr *TrustedTaskRun) SetDefaults(ctx context.Context) {
 }
 
-func (tr *TrustedTaskRun) VerifyTask(
+func (tr *TrustedTaskRun) verifyTaskRun(
 	ctx context.Context,
 	k8sclient kubernetes.Interface,
 	tektonClient versioned.Interface,
 ) (errs *apis.FieldError) {
 	logger := logging.FromContext(ctx)
-	logger.Info("Verifying Resources")
+	logger.Info("Verifying TaskRun")
 
-	if tr.ObjectMeta.Annotations == nil {
-		return apis.ErrMissingField("annotations")
+	if tr.Spec.TaskRef == nil {
+		return nil
 	}
 
-	if tr.ObjectMeta.Annotations[SignatureAnnotation] == "" {
-		return apis.ErrMissingField(fmt.Sprintf("annotations[%s]", SignatureAnnotation))
+	// TODO: ignore cluster task for now, will add later.
+	if tr.Spec.TaskRef.Bundle == "" {
+		return nil
 	}
 
-	signature, err := base64.StdEncoding.DecodeString(tr.ObjectMeta.Annotations[SignatureAnnotation])
-	if err != nil {
-		return apis.ErrGeneric(err.Error(), "metadata")
+	serviceAccountName := os.Getenv("WEBHOOK_SERVICEACCOUNT_NAME")
+	if serviceAccountName == "" {
+		serviceAccountName = "tekton-verify-task-webhook"
 	}
 
-	verifier, err := verifier(ctx, tr.ObjectMeta.Annotations, k8sclient)
-	if err != nil {
-		return apis.ErrGeneric(err.Error(), "metadata")
-	}
+	// TODO: figure out how to config this, one benefit of using this is that we
+	// don't have to configure the feature map.
+	cfg := config.NewStore(logger)
+	cfg.OnConfigChanged(&corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName()},
+		Data: map[string]string{
+			"enable-tekton-oci-bundles": "true",
+		},
+	})
```

Comment:
I'd avoid pulling in the Pipelines config map here - we should keep these separate - we may need to make changes to the upstream fetcher to decouple the TaskResolver from the configmap. For now, you can just construct the config manually and call ToContext without the store.

Reply:
I'm just curious if it is ok to do it in code or via yaml config.
I tried to apply the featuremap yaml to enable this but it doesn't work, is it because I don't include `configmap.Watcher` in current webhook, so it will not watch the config map change?

Reply:
maybe one improvement of doing this in code is that I restore the old `cfg.FeatureFlags.EnableTektonOCIBundles` after I used it here, no matter it is true or false

Reply:
(the new revision of this basically does the same as what I wrote above, so LGTM)